### PR TITLE
Feature refactor

### PR DESCRIPTION
**Pull request by chrisboakes**
- Refactored JS
- Included a load on page init option
- Combined helpers test branch

### CODE REVIEW

File: examples/index.html [hunk 1]
```diff
@@ -9,15 +9,15 @@
     <div class="u-container">
         <img class="c-graphic__image"
             src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkYAAAAAYAAjCB0C8AAAAASUVORK5CYII="
-            data-normal="http://via.placeholder.com/800x600"
+            data-core="http://via.placeholder.com/800x600"
```
Comment (milad-alizadeh) [inline examples/index.html]: should we call this data-src instead?

File: src/js/index.js [hunk 2]
```diff
@@ -1,41 +1,124 @@
...
+
+        if (this.dataNodes.length) {
+            this.setListeners();
+            this.setImageOffsets();
+
+            if (this.options.loadOnScroll) {
+                this.setLoadOnScroll();
+            } else {
+                this.setLoadOnInit();
+            }
```
Comment (milad-alizadeh) [inline src/js/index.js]: `this.options.loadOnScroll ? this.setLoadOnScroll() : this.setLoadOnInit();`

File: src/js/index.js [hunk 3]
```diff
@@ -1,41 +1,124 @@
...
+    getDataNodes() {
+        return helpers.getNodes(`[${this.options.core}]`);
+    }
+
+    /**
+     * Add scroll and resize listeners
+     */
+    setListeners() {
+        if (this.options.loadOnScroll) {
+            window.addEventListener('scroll', helpers.throttle(() => this.setLoadOnScroll(), this.throttle));
```
Comment (milad-alizadeh) [inline src/js/index.js]: I think it should be  `this.options.throttle`

File: src/js/index.js [hunk 4]
```diff
@@ -1,41 +1,124 @@
...
+        }
+
+        window.addEventListener('resize', helpers.debounce(() => this.recalculateAndLoad(), this.resizeDebounce));
```
Comment (milad-alizadeh) [inline src/js/index.js]: I think it should be `this.options.resizeDebounce`

File: src/js/index.js [hunk 5]
```diff
@@ -88,65 +171,20 @@ export default class {
...
      * Recalculate nodes offset
      */
     recalculateAndLoad() {
         this.setImageOffsets();
-        this.setImagesToLoad();
+        if (this.options.loadOnScroll) {
+            this.setLoadOnScroll();
+        } else {
+            this.setLoadOnInit();
+        }
```
Comment (milad-alizadeh) [inline src/js/index.js]: this.options.loadOnScroll ? this.setLoadOnScroll() : this.setLoadOnInit();

File: examples/index.html (hunk 1 above)
Comment (chrisboakes) [reply examples/index.html]: Hmm yeah I think you're right

File: src/js/index.js (hunk 3 above)
Comment (chrisboakes) [reply src/js/index.js]: Good spot!

**Pull request merged**